File: tools/parity/dispatch/pi-agent-kind.ts
// TS dispatch for the pi-agent-kind parity module. The shared TS detector was
// DELETED (the Rust orca-text core is the sole impl — napi in main, wasm in
// the relay), so this adapter drives the napi binding: the vectors' recorded
// goldens now pin the napi surface absolutely. Requires the built addon.

import { requireRustGitBinding } from '../../../src/main/daemon/rust-git-addon'

export function dispatch(fn: string, input: unknown): unknown {
  switch (fn) {
    case 'detectPiAgentKindFromCommand': {
      // TS `undefined` round-trips through JSON as `null`; map it back so the
      // bare-shell (no-command) default case is exercised identically.
      const command = input == null ? undefined : (input as string)
      return requireRustGitBinding().detectPiAgentKindFromCommand(command)
    }
    default:
      throw new Error(`unknown function ${fn}`)
  }
}

File: rust/crates/orca-dispatch/src/modules/pi_agent_kind.rs
//! Parity dispatch for `orca_text::pi_agent_kind` vs
//! `src/shared/pi-agent-kind.ts`.

use orca_text::pi_agent_kind::{detect_pi_agent_kind_from_command, PiAgentKind};
use serde_json::{json, Value};

pub fn dispatch(function: &str, input: &Value) -> Value {
    match function {
        // JSON null (TS `undefined`) and any non-string both yield `as_str() ==
        // None`, matching the TS bare-shell default branch.
        "detectPiAgentKindFromCommand" => {
            Value::String(kind_to_id(detect_pi_agent_kind_from_command(input.as_str())).to_string())
        }
        other => json!({ "__parity_error__": format!("unknown function {other}") }),
    }
}

/// Serialize the enum to its TS string id (`PiAgentKind` = `'pi' | 'omp'`).
fn kind_to_id(kind: PiAgentKind) -> &'static str {
    match kind {
        PiAgentKind::Pi => "pi",
        PiAgentKind::Omp => "omp",
    }
}
